src/test/java/de/hhu/propra/sharingplatform/controller/ItemControllerTest.java

```diff
@@ -0,0 +1,351 @@
+
+    @MockBean
+    private ItemRepo itemRepo;
+
+    @MockBean
+    private UserService userService;
+
+    /*
+    NOT LOGGED in
+     */
+
+    @Test
+    public void itemDetailsNotLoggedIn() throws Exception {
+        mvc.perform(get("/item/details/1000")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().is3xxRedirection());
+    }
+
+    @Test
+    public void getNewItemNotLoggedIn() throws Exception {
+        mvc.perform(get("/item/new")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().is3xxRedirection());
+    }
+
+    @Test
+    public void postNewItemNotLoggedIn() throws Exception {
+        mvc.perform(post("/item/new")
+            .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+            .param("any", "any"))
+            .andExpect(status().is3xxRedirection());
+    }
+
+    @Test
+    public void removeItemNotLoggedIn() throws Exception {
+        mvc.perform(get("/item/remove/1000")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().is3xxRedirection());
+    }
+
+    @Test
+    public void editItemNotLoggedIn() throws Exception {
+        mvc.perform(get("/item/edit/1000")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().is3xxRedirection());
+    }
+
+
+
+    /*
+    LOGGED IN
+     */
+
+    @Test
+    @WithMockUser
+    public void itemDetailsDontExistLoggedIn() throws Exception {
+        mvc.perform(get("/item/details/1000")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    @WithMockUser("accountname")
+    public void itemDetailsExistLoggedInAndOwner() throws Exception {
+        User user = new User();
+        user.setAccountName("accountname");
+        user.setPassword("password");
+        user.setEmail("mail");
+        user.setAddress("address");
+        user.setName("name");
+        user.setBan(false);
+        user.setDeleted(false);
+        user.setId(1l);
+
+        Item item = new Item(user);
+        item.setAvailable(true);
+        item.setBail(1.0);
+        item.setDeleted(false);
+        item.setDescription("desc");
+        item.setLocation("loc");
+        item.setName("item");
+        //item.setOwner(user);
+        item.setPrice(2.0);
+        item.setId(3l);
+
+        Optional<Item> optI = Optional.of(item);
+        Optional<User> optU = Optional.of(user);
+
+        when(itemRepo.findById(3)).thenReturn(optI);
+        //when(userRepo.findByAccountName("accountname")).thenReturn(optU);
+        when(userService.fetchUserByAccountName("accountname")).thenReturn(user);
+        when(userService.fetchUserIdByAccountName("accountname")).thenReturn(1l);
+
+        mvc.perform(get("/item/details/3")
+            .contentType(MediaType.TEXT_HTML))
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("Gegenstand bearbeiten")));

```

Comment:
maybe not the best case to compare because this needs to change to English.
We should use "item name" or other properties of the item